Sources/Hummingbird/Files/FileProvider.swift

```diff
@@ -19,29 +19,31 @@ import NIOPosix
 public protocol FileProvider: Sendable {
     /// File attributes type
     associatedtype FileAttributes
+    /// File identifier
+    associatedtype FileIdentifier
```

Comment:
I understand that this is not needed currently, but i think for the sake of future API flexibility, it's reasonable to require this associated type to be `Hashable`.
For example maybe someday in the future you want to add a new function to the protocol but provide a default implementation to prevent API breakages. `Hashable` could help in a day like that.

Reply:
I cannot see any situation where this would need to be `Hashable` in the protocol. The one situation where this is used currently is the FileMiddleware and that only needs to pass it around to identify a file. If it was used elsewhere in the future we could always add the requirement `where FileProvider.FileIdentifier: Hashable`.

I can see situations where specific implementations of FileProviders would require it to `Hashable` eg A FileProvider that caches files in memory and needs a lookup, but that requirement would be implemented in the implementation and not the protocol.

Reply:
I'm convinced, you're right 🙂.